BlackNectar/BlackNectar/Alerts.swift

```diff
@@ -0,0 +1,251 @@
+//
+//  Alerts.swift
+//  BlackNectar
+//
+//  Created by Cordero Hernandez on 3/20/17.
+//  Copyright © 2017 BlackSource. All rights reserved.
+//
+
+import Archeota
+import AromaSwiftClient
+import Foundation
+import UIKit
+
+extension UIViewController {
+    
+    func presentAlert(_ alert: UIAlertController) {
+        
+        self.present(alert, animated: true, completion: nil)
+    }
+    
+    func showAlert(title: String, message: String) {
+        
+        let alert = UIAlertController(title: title, message: message, preferredStyle: .alert)
+        
+        let ok = UIAlertAction(title: "OK", style: .default, handler: nil)
+        alert.addAction(ok)
+        
+        presentAlert(alert)
+    }
+    
+    func showAlertWithError(message: String) {
+        showAlert(title: "Error", message: message)
+    }
+    
+    func createAlertToSendUserToLocationSettings() -> UIAlertController {
+        
+        let title = "Requesting GPS Access"
+        let message = "Please go to \"Location\" and enable \"While Using the App\""
+        
+        let controller = UIAlertController(title: title, message: message, preferredStyle: .alert)
+        
+        let openSettings = UIAlertAction(title: "Open Settings", style: .default) { _ in
+            self.sendUserToSettings()
+            self.makeNoteThatSendingUserToSettings()
+        }
+        
+        controller.addAction(openSettings)
+        
+        return controller
+    }
+    
+    func sendUserToSettings() {
+        
+        let link = UIApplicationOpenSettingsURLString
+        
+        guard let url = URL(string: link) else {
+            LOG.error("Failed to create URL to \(link)")
+            return
+        }
+        
+        UIApplication.shared.open(url, options: [:], completionHandler: nil)
+        
+    }
+    
+}
+
+//MARK: StoresTableViewController Alert View Code
+extension StoresTableViewController {
+    
+    func sendEmailErrorAlert() {
+        
+        let alert = createAlertForEmailError()
+        self.presentAlert(alert)
+        
+    }
+    
+    func createAlertForEmailError() -> UIAlertController {
+        
+        let title = "Sending Email Failed"
+        let message = "Your device did not successfully send the Email. Please check your wi-fi settings or signal strength and try again."
+        
+        let controller = UIAlertController(title: title, message: message, preferredStyle: .alert)
+        
+        let ok = UIAlertAction(title: "Ok", style: .default) { _ in
+            
+            self.present(self.configureMailComposeViewController(), animated: true, completion: nil)
```

Comment:
You should create and store the mail controller in a variable, instead of doing it here. 
It just feels like too much happening in one line with these two long method calls.